### PR TITLE
Xen troops pv drm offset edid

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: drivers/xen/gntdev-dmabuf.c [hunk 1]
```diff
@@ -198,7 +199,8 @@ static int dmabuf_exp_wait_released(struct gntdev_dmabuf_priv *priv, int fd,
 /* DMA buffer export support. */
 
 static struct sg_table *
-dmabuf_pages_to_sgt(struct page **pages, unsigned int nr_pages)
+dmabuf_pages_to_sgt(struct page **pages, unsigned int nr_pages,
+		    unsigned int data_ofs)
```
Comment (lorc) [inline drivers/xen/gntdev-dmabuf.c:203]: Just to clarify: did you ran `checkpatch.pl` on this series? Formatting there looks odd.

File: drivers/xen/gntdev-dmabuf.c [hunk 2]
```diff
@@ -621,7 +628,7 @@ static struct gntdev_dmabuf *dmabuf_imp_alloc_storage(int count)
 
 static struct gntdev_dmabuf *
 dmabuf_imp_to_refs(struct gntdev_dmabuf_priv *priv, struct device *dev,
-		   int fd, int count, int domid)
+		   int fd, int count, int domid, int *offset)
```
Comment (lorc) [inline drivers/xen/gntdev-dmabuf.c:631]: `unsigned int`, I believe.

File: include/xen/interface/io/displif.h [hunk 3]
```diff
@@ -38,7 +38,7 @@
  *                           Protocol version
  ******************************************************************************
  */
-#define XENDISPL_PROTOCOL_VERSION	"1"
+#define XENDISPL_PROTOCOL_VERSION	1
```
Comment (lorc) [inline include/xen/interface/io/displif.h]: I already mentioned this in Xen PR. We can't just change type of a constant.

File: drivers/gpu/drm/xen/xen_drm_front.c [hunk 4]
```diff
@@ -376,6 +376,60 @@ void xen_drm_front_on_frame_done(struct xen_drm_front_info *front_info,
...
+	memset(&buf_cfg, 0, sizeof(buf_cfg));
+	buf_cfg.xb_dev = front_info->xb_dev;
+	buf_cfg.num_pages = DIV_ROUND_UP(buffer_sz, PAGE_SIZE);
+	buf_cfg.pages = pages;
+	buf_cfg.pgdir = &shbuf;
+	buf_cfg.be_alloc = false;
+
+	ret = xen_front_pgdir_shbuf_alloc(&buf_cfg);
+	if (ret < 0)
+		goto fail_shbuf_alloc;
```
Comment (lorc) [inline drivers/gpu/drm/xen/xen_drm_front.c:402]: just `return ret;`?

File: drivers/xen/gntdev-dmabuf.c (hunk 1 above)
Comment (andr2000) [reply drivers/xen/gntdev-dmabuf.c:203]:
```
static struct sg_table *
dmabuf_pages_to_sgt(struct page **pages, unsigned int nr_pages,
		    unsigned int data_ofs)
```
don't know why, but in the code it is properly formatted:

Screenshot from 2020-05-20 10-46-39

File: drivers/xen/gntdev-dmabuf.c (hunk 2 above)
Comment (andr2000) [reply drivers/xen/gntdev-dmabuf.c:631]: nice catch, fixed

File: drivers/gpu/drm/xen/xen_drm_front.c (hunk 4 above)
Comment (andr2000) [reply drivers/gpu/drm/xen/xen_drm_front.c:402]: indeed ;)

**Comment by arminn** [comment]
@andr2000 It looks like the same for Android merged and tested a long time ago. What about this MR?

**Comment by andr2000** [comment]
> @andr2000 It looks like the same for Android merged and tested a long time ago. What about this MR?

we can probably drop it, but it has 2 patches required for fighting the artifacts (cache flush)

**Comment by arminn** [comment]
" but it has 2 patches required for fighting the artifacts (cache flush)" - you could keep them as a separate branch, but drop MR

**Comment by andr2000** [comment]
So, let it live as is, what's the point in yet another PR?

**Comment by rshym** [comment]
PR is not going to be merged, so will be closed as agreed with the author.

**Pull request closed without merging**